src/helpers/with-page-auth-required.ts

```diff
@@ -62,11 +62,38 @@ export type PageRoute<P, Q extends ParsedUrlQuery = ParsedUrlQuery> = (
  * });
  * ```
  *
+ * If you're using >=Next 12 and {@link getSession} or {@link getAccessToken} without `getServerSideProps`, because you
+ * don't want to require authentication on your route, you might get a warning/error: "You should not access 'res' after
+ * getServerSideProps resolves". You can work around this by wrapping your `getServerSideProps` in
+ * `withPageAuthRequired` using `authRequired: false`, this ensures that the code that accesses `res` will run within
+ * the lifecycle of `getServerSideProps`, avoiding the warning/error eg:
+ *
+ * ```js
+ * // pages/page.js
+ * import { withPageAuthRequired } from '@auth0/nextjs-auth0';
+ *
+ * export default function ProtectedPage({ customProp }) {
+ *   return <div>Protected content</div>;
+ * }
+ *
+ * export const getServerSideProps = withPageAuthRequired({
+ *   authRequired: false,
+ *   async getServerSideProps(ctx) {
+ *     const session = getSession(ctx.req, ctx.res);
+ *     if (session) {
+ *       // user is authenticated
+ *     }
+ *     return { props: { customProp: 'bar' } };
+ *   }
+ * });
+ * ```
+ *
  * @category Server
  */
 export type WithPageAuthRequiredOptions<P = any, Q extends ParsedUrlQuery = ParsedUrlQuery> = {
   getServerSideProps?: GetServerSideProps<P, Q>;
   returnTo?: string;
+  authRequired?: boolean;
```

Comment:
Is an `authRequired` option on `withAuthRequired` the best way to go? It kind of defeats the entire purpose of  `withAuthRequired`. Should we go with a different wrapper instead?

Reply:
I thought about that (eg `withPageAuth` or `withSessionInitialised` etc) - but decided this would be the easiest to grok

Reply:
I'm not sure that should determine the design of this feature (documenting it well along with a good name can make it easy to grok as well). I'll leave it up to you.

Reply:
I went for `getServerSidePropsWrapper` in e523099 (wait until after your day off to take a look :P)